File: Backend/Microservices/Admin/Eefa.Admin.Application.Refactor/UseCases/PersonBankAccounts/Commands/Create/CreatePersonBankAccountCommand.cs
using System.Threading;
using System.Threading.Tasks;
using AutoMapper;
using Eefa.Admin.Application.CommandQueries.PersonBankAccounts.Models;
using MediatR;

public class CreatePersonBankAccountCommand : IRequest<ServiceResult<PersonBankAccountModel>>, IMapFrom<PersonBankAccount>
{
    public int PersonId { get; set; }
    public int? BankId { get; set; }
    public string? BankBranchName { get; set; }
    public int AccountTypeBaseId { get; set; }
    public string AccountNumber { get; set; }
    public string? Description { get; set; }
    public bool IsDefault { get; set; }
    public void Mapping(Profile profile)
    {
        profile.CreateMap<CreatePersonBankAccountCommand, PersonBankAccount>();
    }
}

public class CreatePersonBankAccountCommandHandler : IRequestHandler<CreatePersonBankAccountCommand, ServiceResult<PersonBankAccountModel>>
{
    private readonly IUnitOfWork _unitOfWork;
    private readonly IMapper _mapper;

    public CreatePersonBankAccountCommandHandler(IUnitOfWork unitOfWork, IMapper mapper)
    {
        _unitOfWork = unitOfWork;
        _mapper = mapper;
    }
    public async Task<ServiceResult<PersonBankAccountModel>> Handle(CreatePersonBankAccountCommand request, CancellationToken cancellationToken)
    {
        PersonBankAccount entity = _mapper.Map<PersonBankAccount>(request);

        _unitOfWork.PersonBankAccounts.Add(entity);
        await _unitOfWork.SaveChangesAsync(cancellationToken);

        return ServiceResult.Success(_mapper.Map<PersonBankAccountModel>(entity));
    }
}

File: Frontend/angular-app/src/app/modules/admin/repositories/person-bank-accounts/commands/create-person-bank-account-command.ts
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {PersonBankAccount} from "../../../entities/person-bank-account";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";

export class CreatePersonBankAccountCommand extends IRequest<CreatePersonBankAccountCommand, PersonBankAccount> {

  public personId: number | undefined = undefined;
  public bankId: number | undefined = undefined;
  public bankBranchName: string | undefined = undefined;
  public accountTypeBaseId: number | undefined = undefined;
  public accountNumber: string | undefined = undefined;
  public description: string | undefined = undefined;
  public isDefault: boolean | undefined = undefined;
  constructor() {
    super();
  }

  mapFrom(entity: PersonBankAccount): CreatePersonBankAccountCommand {
    throw new ApplicationError(CreatePersonBankAccountCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  mapTo(): PersonBankAccount {
    throw new ApplicationError(CreatePersonBankAccountCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/admin/personBankAccounts/add";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(CreatePersonBankAccountCommandHandler.name)
export class CreatePersonBankAccountCommandHandler implements IRequestHandler<CreatePersonBankAccountCommand, PersonBankAccount> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService,
  ) {
  }

  async Handle(request: CreatePersonBankAccountCommand): Promise<PersonBankAccount> {
    let httpRequest: HttpRequest<CreatePersonBankAccountCommand> = new HttpRequest<CreatePersonBankAccountCommand>(request.url, request);
    // httpRequest.Query  += ;
    // httpRequest.Headers = httpRequest.Headers.append()

    return await this._httpService.Post<CreatePersonBankAccountCommand, ServiceResult<PersonBankAccount>>(httpRequest).toPromise().then(response => {
      this._notificationService.showSuccessMessage()

      return response.objResult
    })

  }
}
